subworkflows/local/utils_nfcore_phaseimpute_pipeline/main.nf

```diff
@@ -287,9 +305,44 @@ def validateInputParameters() {
     assert params.step, "A step must be provided"
 
     // Check that at least one tool is provided
-    if (params.step.split(',').contains("impute") || params.step.split(',').contains("panelprep")) {
+    if (params.step.split(',').contains("impute")) {
         assert params.tools, "No tools provided"
     }
+
+    // Check that input is provided for all steps, except panelprep
+    if (params.step.split(',').contains("all") || params.step.split(',').contains("impute") || params.step.split(',').contains("simulate") || params.step.split(',').contains("validate")) {
+        assert params.input, "No input provided"
+    }
+
+    // Check that posfile and chunks are provided when running impute only. Steps with panelprep generate those files.
+    if (params.step.split(',').contains("impute") && !params.step.split(',').find { it in ["all", "panelprep"] }) {
+        // Required by all tools except glimpse2
+        if (!params.tools.split(',').contains("glimpse2")) {
+                assert params.posfile, "No --posfile provided for --step impute"
+        }
+        // Required by all tools except STITCH
+        if (!params.tools.split(',').contains("stitch")) {
+                assert params.chunks, "No --chunks provided for --step impute"
+        }
+        // Required by GLIMPSE1 and GLIMPSE2 only
+        if (params.tools.split(',').contains("glimpse")) {
+                assert params.panel, "No --panel provided for imputation with GLIMPSE"
+        }
+
+    // Check that input_truth is provided when running validate
+    if (params.step.split(',').find { it in ["all", "validate"] } ) {
```

Comment:
We should add nf-test for all that.
Maybe in another PR ?

Reply:
I agree!